src/ageml/visualizer.py

```diff
@@ -203,15 +205,16 @@ def age_bias_correction(self, y_true, y_pred, y_corrected, name: str = ""):
         plt.ylabel("Predicted Age")
         plt.xlabel("Chronological Age")
         plt.tight_layout()
-        if name == "":
-            filename = "age_bias_correction.png"
-        else:
-            filename = f"age_bias_correction_{name}.png"
-            plt.suptitle(f"{name}\n", y=1.00)
+        
+        # Save file
+        filename = (f"age_bias_correction"
+                    f"{'_' + tag.covar if tag.covar != '' else ''}"
+                    f"{'_' + tag.system if tag.system != '' else ''}.png")
```

Comment:
NameTag object allows structuring the filename in a modular way, I like it. Put to good use!